src/ui/public/utils/subscribe.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IScope } from 'angular';
+import * as Rx from 'rxjs';
+import { fatalError } from 'ui/notify/fatal_error';
+
+function callInDigest<T extends any[]>($scope: IScope, fn: (...args: T) => void, ...args: T) {
+  try {
+    // this is terrible, but necessary to synchronously deliver subscription values
+    // to angular scopes. This is required by some APIs, like the `config` service,
+    // and beneficial for root level directives where additional digest cycles make
+    // kibana sluggish to load.
+    //
+    // If you copy this code elsewhere you better have a good reason :)
+    if ($scope.$root.$$phase) {
+      fn(...args);
+    } else {
+      $scope.$apply(() => fn(...args));
+    }
+  } catch (error) {
+    fatalError(error);
+  }
+}
+
+/**
+ * Subscribe to an observable at a $scope, ensuring that the digest cycle
+ * is run for subscriber hooks and routing errors to fatalError if not handled.
+ */
+export function subscribe<T>(
```

Comment:
__note:__ `subscribe` is a bit too generic, but I don't know a better name either, the only thing that comes to my mind is `subscribeWithScope`, but it's arguable. So feel free to keep it as is, just wanted to note.

Reply:
I was thinking `subscribeAtScope()`, but then I kept misremembering it as `subscribeWithScope()`, so I figured we can try it out. Shouldn't be a problem to rename it if we desire down the road.